packages/design-system/src/components/section-title.tsx

```diff
@@ -0,0 +1,174 @@
+    color: theme.colors.foregroundMain,
+  },
+  "&:focus-visible::before": {
+    content: "''",
+    position: "absolute",
+    top: 0,
+    bottom: 0,
+    left: theme.spacing[2],
+    right: theme.spacing[2],
+    borderRadius: theme.borderRadius[4],
+    border: `2px solid ${theme.colors.borderFocus}`,
+  },
+  variants: {
+    hasAddIcon: { true: { paddingRight: theme.spacing[16] } },
+  },
+});
+
+const buttonTextStyle = css(textVariants.titles, {
+  overflow: "hidden",
+  textOverflow: "ellipsis",
+  whiteSpace: "nowrap",
+});
+
+const addButtonSlotStyle = css({
+  position: "absolute",
+  right: theme.spacing[6],
+  top: theme.spacing[4],
+});
+
+const dotsSlotStyle = css({
+  display: "flex",
+});
+
+const dotStyle = css({
+  width: theme.spacing[4],
+  height: theme.spacing[4],
+  borderRadius: theme.borderRadius.round,
+  marginRight: -2,
+  variants: {
+    color: {
+      local: { backgroundColor: theme.colors.foregroundLocalFlexUi },
+      remote: { backgroundColor: theme.colors.foregroundRemoteFlexUi },
+    },
+  },
+});
+
+export const SectionTitle = forwardRef(
+  (
+    {
+      isOpen,
+      onOpenChange,
+      onAdd,
+      hasItems = true,
+      addIcon = <PlusIcon />,
+      className,
+      css,
+      children,
+      ...props
+    }: Omit<ComponentProps<"button">, "onClick"> & {
+      isOpen: boolean;
+      onOpenChange: (isOpen: boolean) => void;
+      onAdd?: () => void;
+      /**
+       * If set to `true`, dots aren't shown,
+       * but still affects how isOpen is treated and whether onAdd is called on open.
+       */
+      hasItems?: boolean | Array<"local" | "remote">;
+      addIcon?: ReactNode;
+      css?: CSS;
+    },
+    ref: Ref<HTMLButtonElement>
+  ) => {
+    const isEmpty =
+      hasItems === false || (Array.isArray(hasItems) && hasItems.length === 0);
+
+    const dots = isOpen === false && Array.isArray(hasItems) ? hasItems : [];
+
+    const state = isOpen && isEmpty === false ? "open" : "closed";
+
+    return (
+      <div
+        className={containerStyle({ className, css })}
+        data-state={state}
+        onKeyDown={handleArrowFocus}
+      >
+        <button
+          className={buttonStyle({ hasAddIcon: onAdd !== undefined })}
+          onClick={() => {
+            if (isOpen && isEmpty === false) {
+              onOpenChange(false);
+            }
+            if (isOpen === false) {
+              onOpenChange(true);
+            }
+            if (isEmpty) {
+              onAdd?.();
+            }
+          }}
+          data-state={state}
+          ref={ref}
+          {...props}
+        >
+          <div className={buttonTextStyle()}>{children}</div>
+          {dots.length > 0 && (
+            <div className={dotsSlotStyle()}>
+              {dots.map((color) => (
+                <div key={color} className={dotStyle({ color })} />
+              ))}
+            </div>
+          )}
+        </button>
+        {onAdd && (
```

Comment:
I always don't like logic like this, if we have a handler then we draw UI element.
I would prefer to have here only placeholder for the Add button like its done https://github.com/webstudio-is/webstudio-builder/blob/43617f1a59aa2eb4f9a385ca08b778f69b5f8f24/packages/design-system/src/components/css-value-list-item.tsx#L112

Reply:
The issue is opening logic. 

Also its interesting how to use that `onAdd` in our UI having that we render Section Collapsible elements and style Section in different places. 
How to call onAdd inside Background section if it called on some parent

Reply:
> The issue is opening logic.

Yeah, main reason I've put button inside the component instead of having a `suffix` prop or something is because I wanted to pass the `onClick` callback from the component. And there're also some minor reasons.

> Also its interesting how to use that onAdd in our UI having that we render Section Collapsible elements and style Section in different places.
How to call onAdd inside Background section if it called on some parent
Or we can make SectionTitle components as like as we do custom Sections.

Yeah, we'll have to refactor sections a bit somehow. One options is to make each section render the entire collapsible instead of just its content.